### PR TITLE
feat(PPDSC-2009): add reduced motion to transitionPreset

### PR DESCRIPTION
**Pull request by Xin00163**
PPDSC-2009

- Refactor `getTransitionPresetFromTheme` so that it will render both `no-preference` and `reduce` media queries.
- Change transitionDuration to 0 if motion is reduced

**I have done:**
 - [x] Written unit tests against changes
 - [ ] Written accessibility tests against the component
 - [ ] Written functional tests against the component and/or NewsKit site
 - [ ] Updated relevant documentation
 - [ ] Updated the release notes for the [next release version](https://nidigitalsolutions.jira.com/wiki/spaces/NPP/pages/2349695021/Release%2BNotes%2BCode%2B-%2BWeb%2BDocumentation)
 - [ ] Discussed with QA and/or design about any visual changes

**I have tested manually:**
 - [ ] Renders correctly on IE11
 - [ ] The feature's functionality is working as expected on other browsers such as Safari, Firefox and IE11
 - [ ] There are no accessibility errors reported by WAVE
 - [ ] The screen reader reads the expected text for the tabbable elements.
 - [ ] There are no new errors in the browser console coming from this PR.
 - [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
 - [ ] The Playground feature is working as expected

**Before**

**After:**

**Who should review this PR**

**How to test**

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on: 
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2009-reduced-motion 
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2009-reduced-motion/storybook

**Review by mutebg — CHANGES_REQUESTED** [review]

File: src/utils/__tests__/get-transition-duration.test.ts [hunk 1]
```diff
@@ -1052,4 +1052,298 @@ describe('getTransitionDuration', () => {
...
+      undefined,
+    )({
+      theme,
+    });
+
+    expect(result).toEqual({
+      '@media screen and (max-width: 799px)': {
+        appear: 2000,
+        enter: 2000,
+        exit: 2000,
```
Comment (mutebg) [inline src/utils/__tests__/get-transition-duration.test.ts:1111]: I don't think that's the right behaviour, the way we use getTransitionDuration is inside JS ( React components )  to calculate the Transitions Timeout, so the function should just return a simple JS object or a number.  When you have MQ object it should just return the max values
@pdimova am I correct?

Comment (Xin00163) [reply src/utils/__tests__/get-transition-duration.test.ts:1111]: I think you are right. The returned value is for the timeout which won't recognise MQ

File: src/utils/get-transition-duration.ts [hunk 2]
```diff
@@ -109,14 +111,43 @@ export const getTransitionDurationFromTheme = (
...
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      (acc, [bp, transitionPresetToken]) => {
+        let currentDuration;
+        if (isArrayLikeObject(transitionPresetToken)) {
+          const tokensArray = Object.values(transitionPresetToken) as string[];
+          currentDuration = extractDurationFromPreset(tokensArray, props.theme);
+        } else {
+          const tokensArray = [transitionPresetToken];
+          currentDuration = extractDurationFromPreset(tokensArray, props.theme);
+        }
```
Comment (mutebg) [inline src/utils/get-transition-duration.ts:140]: This is a bit of personal preference, you can write it like so:
```suggestion
      (acc, [, transitionPresetToken]) => {
        const tokensArray: string[] = isArrayLikeObject(transitionPresetToken)
          ? transitionPresetToken
          : [transitionPresetToken];

        const currentDuration = extractDurationFromPreset(
          tokensArray,
          props.theme,
        );
```

**Review by mutebg — DISMISSED** [review]

File: src/utils/get-transition-duration.ts (hunk 2 above)
Comment (Xin00163) [reply src/utils/get-transition-duration.ts:140]: This looks neat 🔥. Thanks!

**Review by mutebg — DISMISSED** [review]

**Review by mutebg — APPROVED** [review]

**Review by baburay23 — APPROVED** [review]

**Pull request merged**